src/main/bluesky/parser.py

```diff
@@ -0,0 +1,55 @@
+import pandas as pd
+import sys
+from bluesky.__main__ import main
+
+flights = pd.read_csv("Flights.csv")
+
+# Remove duplicates from the CALLSIGN column
+flights = flights.drop_duplicates(subset="CALLSIGN", keep="first")
+
+flightCallSign = flights["CALLSIGN"]
+flightProvider = flights["OPERATOR"]
+flightType = flights["ICAO_ACTYPE"]
+flightDest = flights["DEST"]
+flightOrig = flights["ADEP"]
+flightSpeed = flights["TAS"]
+flightAltitude = flights["RFL"]
+flightWeight = flights["TYPE_OF_TRANSPONDER"]
+flightArrival = flights["T0"]
+flightDeparture = flights["T_UPDATE"]
+
+
+def has_required_data(x):
+    # Check if all required columns have non-empty values
+    return all([
+        pd.notna(flightCallSign.iloc[x]) and flightCallSign.iloc[x] != "",
+        pd.notna(flightType.iloc[x]) and flightType.iloc[x] != "",
+        pd.notna(flightOrig.iloc[x]) and flightOrig.iloc[x] != "" and flightOrig.iloc[x] != flightDest.iloc[x] and flightOrig.iloc[x] != "ZZZZ",
+        pd.notna(flightAltitude.iloc[x]) and flightAltitude.iloc[x] != 0,
+        pd.notna(flightSpeed.iloc[x]) and flightSpeed.iloc[x] != 0,
+        pd.notna(flightDest.iloc[x]) and flightDest.iloc[x] != "" and flightDest.iloc[x] != flightOrig.iloc[x] and flightDest.iloc[x] != "ZZZZ"
+    ])
+
+
+def airlines():
```

Comment:
could make 1 line. Should these not return the a array of providers instead of print them?

Reply:
yeah for now i just made a print to see if it all works but they should return